packages/frontend/src/components/UnderlyingData/UnderlyingDataProvider.tsx

```diff
@@ -39,10 +42,61 @@ type UnderlyingDataContext = {
         value: ResultRow[0]['value'],
         meta: TableColumn['meta'],
         row: ResultRow,
+        dimensions?: string[],
+        pivot?: { fieldId: string; value: any },
     ) => void;
+
     closeModal: () => void;
 };
 
+export const getDataFromChartClick = (
+    e: EchartSeriesClickEvent,
+    pivot: string | undefined,
+    explore: Explore,
+) => {
+    const selectedFields = getFields(explore).filter((field) =>
+        e.dimensionNames.includes(getFieldId(field)),
+    );
+    const selectedDimensions = getDimensions(explore).filter((dimension) =>
+        e.dimensionNames.includes(getFieldId(dimension)),
+    );
+
+    const selectedField =
+        selectedDimensions.length > 0
+            ? selectedDimensions[0]
+            : selectedFields[0];
+    const selectedValue = e.data[getFieldId(selectedField)];
+    const row: ResultRow = Object.entries(e.data as Record<string, any>).reduce(
+        (acc, entry) => {
+            const [key, val] = entry;
+            return { ...acc, [key]: { value: { raw: val, formatted: val } } };
+        },
+        {},
+    );
+
+    const getPivotValue = () => {
+        const pivotField = getDimensions(explore).find(
+            (dimension) => getFieldId(dimension) === pivot,
+        );
+        switch (pivotField?.type) {
+            case DimensionType.BOOLEAN:
+                return e.seriesName === 'Yes' ? true : false;
+            default:
+                return e.seriesName;
+        }
+    };
```

Comment:
I think this will fail when the values are formatted e.g dates, numbers in the thousands, currency, rounded numbers, etc. I don't think you have access to the raw pivot value in echarts 🤔 
I think we need to change `getPivotSeries` in `useEcharts.ts` to have the pivot information from `series.encode.yRef.pivotValues`. This way you would have access to the pivot field id and the raw pivot value.

Reply:
yeah, I'm only fixing booleans here, but we'll have to take the raw for other formats. Not sure if it is worth doing it as a separate PR

Reply:
I think we can work on this on a following ticket if @TuringLovesDeathMetal doesn't mind having this limitation/bug for a short time.

Reply:
How about this ? https://github.com/lightdash/lightdash/pull/2957/commits/64411bd5975a060cd610e04272640eead01a49cf

Reply:
yep, I'm okay with temporary bugs :)